contracts/src/TeleporterTokenDestination.sol

```diff
@@ -37,11 +49,37 @@ abstract contract TeleporterTokenDestination is
     address public immutable feeTokenAddress;
 
     /**
-     * @notice Fixed gas cost for performing a multihop transfer on the `sourceBlockchainID`
-     * , before forwarding to the final destination bridge instance.
+     * @notice tokenMultiplier allows this contract to scale the number of tokens it sends/receives to/from
+     * the source chain.
+     *
+     * @dev This can be used to normalize the number of decimals places between the tokens on
+     * the two subnets. Is calculated as 10^d, where d is decimalsShift specified in the constructor.
+     */
+    uint256 public immutable tokenMultiplier;
+
+    /**
+     * @notice If multiplyOnReceive is true, the raw token amount value will be multiplied by `tokenMultiplier` when tokens
+     * are transferred from the source chain into this destination chain, and divided by `tokenMultiplier` when
+     * tokens are transferred from this destination chain back to the source chain. This is intended
+     * when the "decimals" value on the source chain is less than the native EVM denomination of 18.
+     * If multiplyOnReceive is false, the raw token amount value will be divided by `tokenMultiplier` when tokens
+     * are transferred from the source chain into this destination chain, and multiplied by `tokenMultiplier` when
+     * tokens are transferred from this destination chain back to the source chain.
+     */
+    bool public immutable multiplyOnReceive;
+
+    /**
+     * @notice Fixed gas cost for performing a multihop transfer on the `sourceBlockchainID`,
+     * before forwarding to the final destination bridge instance.
      */
     uint256 public constant MULTIHOP_REQUIRED_GAS = 220_000;
 
+    /**
+     * @notice The amount gas added to the required gas limit for a multihop call message
+     * for each byte of the recipient payload.
+     */
+    uint256 public constant MULTIHOP_CALL_GAS_PER_BYTE = 1_000;
```

Comment:
Any constant that defines gas cost on an external chain makes an assumption about that chain's VM gas rules. I'm not sure to what degree that's customizable in `subnet-evm`, but I'm wondering if this has been taken into account.

Reply:
I definitely didn't check this value super closely. I'll create an issue to better estimate/evaluate it. 

I do think a constant here is still sufficient, since it is referring to the gas cost of an interaction with a specific contract on the specific source chain for the given destination, but we should more properly evaluate what the correct constant value is for sure.

Reply:
Depending on if and how much we're expecting these gas costs to vary chain to chain, it may make sense to bundle them together with the `sourceBlockchainID` in a struct as a constructor argument.

Reply:
Created an issue here, and noted the possibly of moving them to constructor arguments.

https://github.com/ava-labs/teleporter-token-bridge/issues/82